test/junit/scala/collection/StrictOptimizedSeqTest.scala

```diff
@@ -20,4 +20,24 @@ class StrictOptimizedSeqTest {
 
     assertEquals(Vector("a", "aa", "aaa", "bbbb"), result)
   }
+
+  @Test
+  def `empty intersection has correct component type for array`(): Unit = {
+    val something = Array(3.14)
+    val nothing   = Array[Double]()
+    val empty     = Array.empty[Double]
+
+    assertEquals(classOf[Double], nothing.intersect(something).getClass.getComponentType)
```

Comment:
I think we should remove the classOf asserts because I think they are stronger than the contract really is.

Reply:
I just assumed someone working with an array of primitive double always wants that. Ichoran always complains if something is unexpectedly boxed and performance suffers. or he used to complain. I'll give it some naive thought.

Reply:
It's a good point that it's stronger than the contract.  But I agree it's good to have the assertion to track the behaviour nonetheless.

Reply:
`Array` says I get back an array of the same type. I don't know what `A` is in the scaladoc, but I hope it is `T`.
```
def intersect[B >: A](that: collection.Seq[B]): Array[T]
```